zetaclient/utils.go

```diff
@@ -119,6 +124,94 @@ func DepositorFee(satPerByte int64) float64 {
 	return float64(satPerByte) * float64(BtcOutTxBytesDepositor) / satoshiPerBitcoin
 }
 
+// CalcBlockAvgFeeRate calculates the average gas rate (in sat/vByte) for a given block
```

Comment:
I think you could sync with @kevinssgh on this, with https://github.com/zeta-chain/node/pull/1640

This can be put in `bitcoin` package

Reply:
I'll reach out to him. The two open PRs will have conflict to solve.